src/pudl_rmi/connect_deprish_to_ferc1.py

```diff
@@ -650,3 +652,141 @@ def _allocate_col(
         .rename(columns={output_col: allocate_col})
     )
     return to_allocate.loc[:, [allocate_col]]
+
+
+##################
+# Validation Tests
+##################
+
+# This whole section should pprrrrooobably be moved into a validation layer of
+# CI tests for this repo.
+
+def test_consistency_of_data_stages(df1, df2):
+    """
+    Test the consistency of two stages of the depreciation data processing.
+
+    The data that is processed in this repo goes along multiple stages of its
+    journey. This function right now is hard coded to test two the depreication
+    data's main data columns. Right now, this is hard coded to fail when there
+    are more inconsitent plants and utilities than are currently known.
+
+    TODO: I think :func:``data_col_test`` is a perfect candidate for the
+    decorators that let you test multiple inputs (different stages of the data
+    and different data columns) as well as output values (in this case the
+    number of known bad utilities and plants).
+
+    Args:
+        df1: One dataframe to sum and check consistency with ``df2``. Both
+            dfs should have columns ``data_col`` as well as identifying
+            columns: ``['report_date', 'data_source', 'utility_id_pudl',
+            'plant_id_eia']``
+        df2: Other dataframe to sum and check consistency against ``df1``.
+
+    """
+    util_bad, plant_bad = data_col_test(
+        df1=df1, df2=df2, data_col='plant_balance_w_common')
+
+    # known baddies. We need to track these down
+    assert(len(plant_bad) <= 53)
+    assert(len(util_bad) <= 88)
+
+    util_bad, plant_bad = data_col_test(
+        df1=df1, df2=df2, data_col='unaccrued_balance_w_common')
+
+    # known baddies. We need to track these down
+    assert(len(plant_bad) <= 35)
+    assert(len(util_bad) <= 70)
+
+
+def data_col_test(df1, df2, data_col: str) -> pd.DataFrame:
+    """
+    Check consistency of column at the plant and utility level in two inputs.
+
+    Args:
+        df1: One dataframe to sum and check consistency with ``df2``. Both
+            dfs should have columns ``data_col`` as well as identifying
+            columns: ``['report_date', 'data_source', 'utility_id_pudl',
+            'plant_id_eia']``
+        df2: Other dataframe to sum and check consistency against ``df1``.
+        data_col: data column to check. Column must be in both ``df1`` and
+            ``df2``.
+
+    """
+    util_test = gb_test(
+        df1,
+        df2,
+        data_col=data_col,
+        by=['report_year', 'data_source', 'utility_id_pudl']
```

Comment:
So in the parameterization, it seems like we would iterate over
* aggregation columns (utility vs plant level aggregation)
* data columns (with their attendant catalogs of known bad records)
* different combinations of stages within the process

Is that right?

Reply:
I think the aggregation columns should always be utility and plant. At least in this repo's universe that feels true. I could imagine varying this in other contexts, but here it'll always be those.

I also believe the known baddies will need to be a part of this parameterization.

Reply:
The known baddies is what I was trying to get at in the 2nd point -- that there's a catalog of known bad records associated with each data column. Are you saying that they need to be managed in some other way too? Does every combination of stages and aggregation column (utility or plant) have it's own distinct set of bad records? That seems complicated.

Reply:
This is vv out of date over here bc everything got moved/rearranged. But to your last question... yes. Every stage is a different processing step and I want to know if things change. I believe a large factor in these baddies is unconnected or poorly connected records... but I want to test at each stage. The depreciation data goes through 3 stages so this is currently what is being tested:
- stage 1 compared to stage 2 (deprish vs deprish_eia)
- stage 2 compared to stage 3 (deprish_eia vs deprish_ferc1)
-  and stage 1 compared to stage 3 (deprish vs deprish_ferc1)
 
The last one is slightly duplicative but it is what really matters in the context of the final outputs (stage 3 in  this case is the final outputs).

Right now it is setup to test just one column from the depreciation data and one column from the ferc1 side. The FERC1 data only goes through two stages (#1: ferc1_eia and #2: deprish_ferc1) so it is a little simpler. We've discussed that in the future it might be nice to enable testing multiple columns from each of these original datasets at the same time, but this is oos for this PR.